src/components/Spinner/Spinner.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import styled, { keyframes } from "styled-components";
+import PanIcon from "./PanIcon";
+import PancakeIcon from "./PancakeIcon";
+import { SpinnerProps } from "./types";
+
+const rotate = keyframes`
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+`;
+
+const float = keyframes`
+	0% {
+		transform: translatey(0px);
+	}
+	50% {
+		transform: translatey(-20px);
+	}
+	100% {
+		transform: translatey(0px);
+	}
+`;
+
+const Container = styled.div`
+  position: relative;
+`;
+
+const RotatingPancakeIcon = styled(PancakeIcon)`
+  position: absolute;
+  top: 0;
+  left: 0;
+  animation: ${rotate} 2s linear infinite;
```

Comment:
Might want to consider making the duration customizable in the future.

Reply:
I think the duration should be decided by the designer only